app/models/user.rb

```diff
@@ -42,4 +48,16 @@ def sunetid
   def to_honeybadger_context
     { user_id: id, user_email: email }
   end
+
+  sig { returns(T::Boolean) }
+  def agreed_to_terms_recently?
+    return false unless last_work_terms_agreement
+
+    terms_agreement_renewal_timeframe < last_work_terms_agreement
+  end
+
+  sig { returns(T::Boolean) }
+  def agreed_to_terms
+    update(last_work_terms_agreement: Time.zone.now)
+  end
 end
```

Comment:
when the user agrees to the terms, we need to update the current time in the database